stix2/base.py

```diff
@@ -124,11 +124,20 @@ def __len__(self):
 
     # Handle attribute access just like key access
     def __getattr__(self, name):
-        return self.get(name)
+        # If the requested attribute is not None return its value
```

Comment:
The "is the key there" check could be `if name in self` (or equivalently, `if name in self._inner`).  Using the `in` operator implicitly leverages the dict/key access mechanism.  So the whole solution could be as simple as:
```python
if name in self:
    return self[name]
raise AttributeError(name)
```
Or just attempt the access and catch KeyError, but I prefer to only use the exception mechanism to handle exceptional situations.  (Which admittedly goes against some of Python's fundamental designs, e.g. raising StopIteration when an iterator is exhausted.  Reaching the end of iteration is a completely normal occurrence.)

If dict and attribute access are to behave identically, then `bundle.objects` *must* cause an AttributeError if the `objects` attribute is not set, because that's what dict/key access would do (except with a KeyError).  I've been operating under the assumption that's what you wanted.  That's what the [comment](https://github.com/emmanvg/cti-python-stix2/blob/a38b912d1960245b84c5cb9c9af16fde8e420ac5/stix2/base.py#L125) in the code said.  If so, then if users want to check for property existence via the dict/key mechanism, they can use `if "some_prop" in my_stix_obj`.  If they want to check via attributes, they can use `if hasattr(my_stix_obj, "some_prop")`.  Using `if my_stix_obj.some_prop` as an existence check would be incorrect.